pkg/controller/controller_binding.go

```diff
@@ -301,71 +300,16 @@ func (c *controller) reconcileServiceBinding(binding *v1beta1.ServiceBinding) er
 			return nil
 		}
 
-		var (
-			parameters                 map[string]interface{}
-			parametersChecksum         string
-			rawParametersWithRedaction *runtime.RawExtension
+		parameters, parametersChecksum, rawParametersWithRedaction, err := c.prepareInProgressProperties(
+			instance,
```

Comment:
Change `instance` to `binding`. 

You should remove the first parameter entirely from the `prepareInProgressProperties` method. For recording the event, the cloned copy of the resource can be used instead of the original.